src/D3MHub.sol

```diff
@@ -239,32 +238,31 @@ contract D3MHub {
         emit Unwind(ilk, amount);
     }
 
-    function _fullUnwind(bytes32 ilk, ID3MPool _pool, address urn) internal {
+    function _unwind(bytes32 ilk, ID3MPool _pool, address urn) internal {
         uint256 amount = _pool.maxWithdraw();
         if (amount > 0) {
             uint256 toSlip = _min(vat.gem(ilk, urn), amount);
             require(toSlip <= MAXINT256, "D3MHub/overflow");
+            vat.slip(ilk, urn, -int256(toSlip));
 
             _pool.withdraw(amount);
             daiJoin.join(address(this), amount);
-
-            vat.slip(ilk, urn, -int256(toSlip));
             vat.move(address(this), vow, amount * RAY);
         }
         emit Unwind(ilk, amount);
     }
 
     function _normal(bytes32 ilk, ID3MPool _pool, uint256 Art, uint256 lineWad) internal {
```

Comment:
Why is `Art` being passed in if it's not being used?

Reply:
It is being used in the `require`.

Reply:
Sure, but it's just a bit weird to pass something in just to make sure it's the same as another value. I guess it saves gas..

Could maybe think about just removing that check. I'm not sure if it helps us all that much.

Reply:
Well the problem is otherwise the code is conceptually wrong. We would be comparing ilk debt ceiling against one `urn`'s debt. That's was the main reason to actually use `Art` in the first place, or `art` as now but with the condition that equals to `Art`.

Reply:
It is a bit annoying from the code simplicity perspective, I agree, but this way leaves me with a better mental peace.

Reply:
Ok, I don't feel strongly. You can leave as-is.

Reply:
Apologies for necromancing, but is there any chance we would ever possibly have two D3Ms for the same ilk going at once, e.g. as part of a code migration? I figure probably not, worst case you just deploy a new `ilk`, but wanted to float the idea out there.

Reply:
I couldn't find the usage, as migration should always be able to happen atomically. And in the worst case, as you said, we can deploy with a new `ilk`. It felt much better to keep that invariant in the code.

Reply:
sg resolving